backend/api/v2beta1/recurring_run.proto

```diff
@@ -0,0 +1,239 @@
+  string experiment_id = 6;
+}
+
+message ListRecurringRunsResponse {
+  // A list of recurring runs returned.
+  repeated RecurringRun recurringRuns = 1;
+
+  // The total number of recurring runs for the given query.
+  int32 total_size = 2;
+
+  // The token to list the next page of recurring runs.
+  string next_page_token = 3;
+}
+
+message EnableRecurringRunRequest {
+  // The ID of the recurring runs to be enabled.
+  string rec_run_id = 1;
+}
+
+message DisableRecurringRunRequest {
+  // The ID of the recurring runs to be disabled.
+  string rec_run_id = 1;
+}
+
+message DeleteRecurringRunRequest {
+  // The ID of the recurring run to be deleted.
+  string rec_run_id = 1;
+}
+
+// CronSchedule allow scheduling the recurring run with unix-like cron.
+message CronSchedule {
+  // The start time of the cron job.
+  google.protobuf.Timestamp start_time = 1;
+
+  // The end time of the cron job.
+  google.protobuf.Timestamp end_time = 2;
+
+  // The cron string. For details how to compose a cron, visit
+  // ttps://en.wikipedia.org/wiki/Cron
+  string cron = 3;
+}
+
+// PeriodicSchedule allow scheduling the recurring run periodically with certain interval.
+message PeriodicSchedule {
+  // The start time of the periodic recurring run.
+  google.protobuf.Timestamp start_time = 1;
+
+  // The end time of the periodic recurring run.
+  google.protobuf.Timestamp end_time = 2;
+
+  // The time interval between the starting time of consecutive recurring runs.
+  int64 interval_second = 3;
+}
+
+// Trigger defines what starts a pipeline run.
+message Trigger {
+  oneof trigger {
+    CronSchedule cron_schedule = 1;
+    PeriodicSchedule periodic_schedule = 2;
+  }
+}
+
+message RecurringRun {
+  // Output. Unique run ID generated by API server.
+  string rec_run_id = 1;
+
+  // Required input field. Recurring run name provided by user. Not unique.
+  string display_name = 2;
+
+  // Optional input field. Describes the purpose of the recurring run.
+  string description = 3;
+
+  // Required input field. Specifies the source of the pipeline spec for this 
+  // recurring run. Can be either a pipeline id, or a pipeline spec.
+  oneof pipeline_source {
+    // The ID of the pipeline user uploaded before.
+    string pipeline_id = 4;
+    // The pipeline spec.
+    google.protobuf.Struct pipeline_spec = 5;
+  }
+
+  // Runtime config of the pipeline. 
+  RuntimeConfig runtime_config = 6;
+
+  // Optional input field. Specifies which Kubernetes service account this recurring run uses.
+  string service_account = 7;
+
+  // Required input field.
+  // Specifies how many runs can be executed concurrently. Range [1-10].
+  int64 max_concurrency = 8;
+
+  // Required input field.
+  // Specifies how a run is triggered. Support cron mode or periodic mode.
+  Trigger trigger = 9;
+
+  // Required input.
+  enum Mode {
+    MODE_UNSPECIFIED = 0;
+    MODE_ENABLED = 1;
```

Comment:
It turned out all the `ENUM`s in the same message have to be unique. That's why I added the `MODE_` back.

Reply:
Did some research, the protobuf doc actually showcase enum with name prefix for all its values:

https://developers.google.com/protocol-buffers/docs/proto3#enum

Read [AIP-126](https://google.aip.dev/126) again, although it suggests 

> The other values should not be prefixed by the name of the enum itself

It also calls out specifically the reason you mentioned: 

> Note that some languages (including C++) hoist enum values into the parent namespace, which can result in conflicts for enums with the same values in the same proto package. To avoid this, multiple enums in the same proto package must not share any values. To avoid sharing values, APIs may prefix enum values with the name of the enum. In this case, they must do so consistently within the enum.

So I think we should probably keep it consistent not only within the enum but across our entire code base. Meaning we should probably go back change the storage state enum, that way we also avoid all the conversion logic you added for v1/v2 compatibility. WDYT?